android/expoview/src/main/java/host/exp/exponent/ExpoUpdatesAppLoader.kt

```diff
@@ -399,10 +399,35 @@ class ExpoUpdatesAppLoader @JvmOverloads constructor(
       return headers
     }
 
+  private val isRunningOnEmulator: Boolean
+    get() = Build.FINGERPRINT.startsWith("generic")
+      || Build.FINGERPRINT.startsWith("unknown")
+      || Build.MODEL.contains("google_sdk")
+      || Build.MODEL.lowercase(Locale.ROOT).contains("droid4x")
+      || Build.MODEL.contains("Emulator")
+      || Build.MODEL.contains("Android SDK built for x86")
+      || Build.MANUFACTURER.contains("Genymotion")
+      || Build.HARDWARE.contains("goldfish")
+      || Build.HARDWARE.contains("ranchu")
+      || Build.HARDWARE.contains("vbox86")
+      || Build.PRODUCT.contains("sdk")
+      || Build.PRODUCT.contains("google_sdk")
+      || Build.PRODUCT.contains("sdk_google")
+      || Build.PRODUCT.contains("sdk_x86")
+      || Build.PRODUCT.contains("vbox86p")
+      || Build.PRODUCT.contains("emulator")
+      || Build.PRODUCT.contains("simulator")
+      || Build.BOARD.lowercase(Locale.ROOT).contains("nox")
+      || Build.BOOTLOADER.lowercase(Locale.ROOT).contains("nox")
+      || Build.HARDWARE.lowercase(Locale.ROOT).contains("nox")
+      || Build.PRODUCT.lowercase(Locale.ROOT).contains("nox")
+      || Build.SERIAL.lowercase(Locale.ROOT).contains("nox")
+      || (Build.BRAND.startsWith("generic") && Build.DEVICE.startsWith("generic"))
+
   private val clientEnvironment: String
     get() = if (Constants.isStandaloneApp()) {
```

Comment:
This code is already calling into Constants -- possible to ask Constants if the app is running in an emulator?

Reply:
That's the Constants class inside of expoview, not expo-constants.